arch/arm/core/aarch32/cortex_m/tz/secure_entry_functions.ld

```diff
@@ -5,20 +5,23 @@
  */
 
 /* nRF-specific defines. */
-#ifdef CONFIG_CPU_HAS_NRF_IDAU
+#if defined(CONFIG_CPU_HAS_NRF_IDAU) && CONFIG_ARM_NSC_REGION_BASE_ADDRESS == 0
 	/* This SOC needs the NSC region to be at the end of an SPU region. */
+	#define __NSC_ALIGN (ALIGN(CONFIG_NRF_SPU_FLASH_REGION_SIZE) \
```

Comment:
@oyvindronningstad this does not only fix the bug of the re-definition, but also fixes a second bug and prevents the location counter to move back, that's correct?

Reply:
Correct, they are in two separate commits.

Reply:
thanks